dm/dm-safe-mode.md

```diff
@@ -0,0 +1,100 @@
+
+By rewriting SQL statements, before duplicate insert or update operations, DM uses the new row data to overwrite the existing row data. This guarantees that insert and update operations are executed repeatedly.
+
+## Enable safe mode
+
+You can enable safe mode either automatically or manually. This section describes the detailed steps.
+
+### Automatically enable
+
+When DM resumes an incremental replication task from a checkpoint (DM worker restart or network reconnection), DM automatically enables safe mode for a period.
+
+Whether to enable safe mode is related to `safemode_exit_point` in the checkpoint. When an incremental replication task is paused abnormally, DM replicates all DML statements in the memory to the downstream and records the latest binlog position in the memory pulled from the upstream as `safemode_exit_point`, which is saved in the last checkpoint before the abnormal pause.
+
+When resuming an incremental replication task from the checkpoint, DM determines whether to enable safe mode based on the following logic:
+
+- If the checkpoint contains `safemode_exit_point`, the incremental replication task is paused abnormally. When DM resumes the task, if DM detects that the binlog position of the checkpoint to be resumed is earlier than `safemode_exit_point`, the binlog events between the checkpoint and the `safemode_exit_point` might have been processed in the downstream. After the task is resumed, some binlog events are executed repeatedly. Therefore, DM determines that safe mode should be enabled for these binlog positions. After the binlog position exceeds the `safemode_exit_point`, if safe mode is not manually enabled, DM automatically disables safe mode.
+
+- If the checkpoint does not contain `safemode_exit_point`, there are two cases:
+
+    1. This is a new task, or this task is paused as expected.
+    2. This task is paused abnormally but DM fails to record `safemode_exit_point`, or the DM process exits abnormally.
+
+    In the second case, DM does not know which binlog events after the checkpoint are executed in the downstream. To ensure that repeatedly executed binlog events do not cause any problems, DM automatically enables safe mode between the preceding two checkpoints. The default interval between two checkpoints is 30 seconds, which means when a normal incremental replication task starts, safe mode is enforced for the first 60 seconds (2 * 30 seconds).
+
+    You can change the checkpoint interval by setting the `checkpoint-flush-interval` item in syncer configuration, thereby adjusting the safe mode period at the beginning of a incremental replication task. It is not recommended to adjust this setting. If necessary, you can [manually enable safe mode](#manually-enable).
```

Comment:
```suggestion
    Usually, it is not recommended to change the checkpoint interval to adjust the safe mode period at the beginning of the incremental replication task. However, if you do need a change, you can [manually enable safe mode](https://github.com/pingcap/docs/pull/8996#manually-enable) (recommended) or set the `checkpoint-flush-interval` item in syncer configuration.
```
Made a minor edit by wrapping "recommended" in parenthesis. @shichun-0415  PTAL

Reply:
> Made a minor edit by wrapping "recommended" in parenthesis. @shichun-0415 PTAL

LGTM